static/app/components/events/interfaces/spans/spanBar.tsx

```diff
@@ -216,6 +217,22 @@ class SpanBar extends React.Component<SpanBarProps, SpanBarState> {
     if (this.spanRowDOMRef.current) {
       this.connectObservers();
     }
+    this.scrollIntoView();
+  }
+
+  componentDidUpdate(prevProps: SpanBarProps) {
+    const {location: prevLocation} = prevProps;
+    const {location} = this.props;
+
+    // This has the caveat that clicking the same anchor twice
+    // in a row will not link to it the second time.
+    //
+    // e.g. Click the anchor on 1 span, scroll around and click
+    // the same anchor again. Only the first click will scroll
+    // the span to the top, the second one is a no op.
+    if (prevLocation.hash !== location.hash) {
```

Comment:
@Zylphrex we can attach an event listener for whenever the hash changes: https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onhashchange

```
window.addEventListener("hashchange", this.scrollIntoView, false);
```